corehq/apps/locations/bulk_management.py

```diff
@@ -548,6 +544,29 @@ def _validate_geodata(self):
         ]
 
     @memoized
+    def _check_required_locations_missing(self):
+        if not self.locations_to_be_deleted or not self.old_collection:
+            # skip this check if no old locations or no location to be deleted
+            return []
+
+        old_locs_by_parent = self.old_collection.locations_by_parent_code
+
+        missing_locs = []
+        listed_sites = {l.site_code for l in self.all_listed_locations}
+        for loc in self.locations_to_be_deleted:
+            required_locs = old_locs_by_parent[loc.site_code]
+            missing = set([l.site_code for l in required_locs]) - listed_sites
+            if missing:
+                missing_locs.append((missing, loc))
+
+        return [
+            _(u"Location '{code}' in sheet '{type}' at index {index} is being deleted, so all its "
+              "child locations must be present in the upload, but child locations '{locs}' are missing")
```

Comment:
Do all the child locations just have to be listed, or do they also have to be deleted or reassigned? I would say that in this message if so.